src/gfx/reverse.cpp

```diff
@@ -241,24 +253,34 @@ void reverse() {
 
 	if (tilemap) {
 		if (attrmap) {
-			for (auto [id, attr] : zip(*tilemap, *attrmap)) {
-				bool bank = attr & 1 << 3;
-				if (id >= options.maxNbTiles[bank]) {
+			for (size_t index = 0; index < nbTileInstances; ++index) {
+				uint8_t tileID = (*tilemap)[index];
+				uint8_t attr = (*attrmap)[index];
+				if (bool bank = attr & 1 << 3; tileID >= options.maxNbTiles[bank]) {
+					size_t tx = index % width, ty = index / width;
 					warning(
-					    "Tile #%" PRIu8 " was referenced, but the limit for bank %u is %" PRIu16,
-					    id,
+					    "Tilemap references tile #%" PRIu8
+					    " at (%zu, %zu), but the limit for bank %u is %" PRIu16,
+					    tileID,
+					    tx,
+					    ty,
 					    bank,
 					    options.maxNbTiles[bank]
 					);
 				}
 			}
 		} else {
-			for (auto id : *tilemap) {
-				if (id >= options.maxNbTiles[0]) {
-					warning(
-					    "Tile #%" PRIu8 " was referenced, but the limit is %" PRIu16,
-					    id,
-					    options.maxNbTiles[0]
+			size_t limit =
+			    std::min<size_t>(nbTiles, options.maxNbTiles[0]) + options.baseTileIDs[0];
+			for (size_t index = 0; index < nbTileInstances; ++index) {
+				if (uint8_t tileID = (*tilemap)[index]; tileID >= limit) {
+					size_t tx = index % width, ty = index / width;
+					fatal(
```

Comment:
Rather than terminating at the first violation, I'd like this to be an `error`, and after the loop, `if (nbErrors != 0) { giveUp(); }`